src/main/java/seedu/address/ui/ShowPersonWindow.java

```diff
@@ -0,0 +1,117 @@
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.ClipboardContent;
+import javafx.scene.layout.FlowPane;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.person.Person;
+
+/**
+ * Controller for a Show Person Details page
+ */
+public class ShowPersonWindow extends UiPart<Stage> {
+
+    private static final String FXML = "ShowPersonWindow.fxml";
+
+    @FXML
+    private Label name;
+    @FXML
+    private Label id;
+    @FXML
+    private Label phone;
+    @FXML
+    private Label email;
+    @FXML
+    private Label address;
+    @FXML
+    private FlowPane subjects;
+    @FXML
+    private FlowPane tags;
+
+    /**
+     * Creates a new ShowPersonWindow.
+     *
+     * @param root Stage to use as the root of the ShowPersonWindow.
+     */
+    public ShowPersonWindow(Stage root) {
+        super(FXML, root);
+    }
+
+    /**
+     * Sets the Details of the person to be shown.
+     *
+     * @param person The person whose details are to be shown.
+     */
+    public void setPersonDetails (Person person) {
+        name.setText(person.getName().fullName);
+        phone.setText(person.getPhone().value);
+        address.setText(person.getAddress().value);
+        email.setText(person.getEmail().value);
+        person.getTags().stream()
+                .sorted(Comparator.comparing(tag -> tag.tagName))
+                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+        person.getSubjects().stream()
+                .sorted(Comparator.comparing(subject -> subject.subjectName))
+                .forEach(subject -> subjects.getChildren()
+                        .add(new ColoredTextEntry(subject.subjectName.toString(), subject.getColour())));
+    }
+
+    /**
+     * Creates a new ShowPersonWindow.
+     */
+    public ShowPersonWindow() {
+        this(new Stage());
+    }
+
+    /**
+     * Shows the Show Person window.
+     * @throws IllegalStateException
+     *     <ul>
+     *         <li>
+     *             if this method is called on a thread other than the JavaFX Application Thread.
+     *         </li>
+     *         <li>
+     *             if this method is called during animation or layout processing.
+     *         </li>
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        getRoot().show();
+        getRoot().centerOnScreen();
+    }
+
+    /**
+     * Returns true if the Show Person window is currently being shown.
+     */
+    public boolean isShowing() {
+        return getRoot().isShowing();
+    }
+
+    /**
+     * Hides the Show Person window.
+     */
+    public void hide() {
+        getRoot().hide();
+    }
+
+    /**
+     * Focuses on the Show Person window.
+     */
+    public void focus() {
```

Comment:
I am a little bit concern with this; will this block further commands the user want to input to main windows? and how to quit this windows with ease using command line?

Reply:
Some reference:
"The stage with focus is the one that will respond to user input. If you click inside a particular stage, it typically receives focus, and its controls become active and responsive to user input. Other unfocused stages will not respond to input until you click or interact with them to bring them into focus.This behavior allows you to create multi-window applications in JavaFX where each stage can have its own set of controls and user interactions, and the user can switch between them by clicking on the respective stage or window. It's common in many desktop applications to have multiple windows or dialog boxes open simultaneously, and JavaFX's focus system helps manage user interactions with these windows."

Reply:
But still, I believe this is a mild enhancement issue, not very severe

Reply:
The Show Person Window will not block further input to the main window. However the issue of quitting from the command line is a good point. I will try to change the window into a panel in the main window.